include/osquery/killswitch.h

```diff
@@ -0,0 +1,55 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <boost/core/noncopyable.hpp>
+#include <osquery/expected.h>
+#include <osquery/status.h>
+#include <string>
+
+namespace osquery {
+
+class Killswitch : private boost::noncopyable {
+ private:
+  Killswitch();
+
+ public:
+  ~Killswitch();
+
+  /*
+  // Author: @guliashvili
+  // Creation Time: 11 / 07 / 2018
+  bool isMyTutorialFeatureEnabled();
+  */
+
+  static Killswitch& get() {
+    static Killswitch killswitch;
+    return killswitch;
+  }
+
+  Status refresh();
+
+ private:
+  bool isNewCodeEnabled(const std::string& key);
+
+  enum class IsEnabledError {
```

Comment:
As far as this list is a list of `Killswitch` errors you could you a more generic name for it. Smth like: `Error`. And usage of this class would look like:
```c++
return createError(
  Killswitch::Error::IncorrectResponseFormat,
  "isEnabled key missing in response of the action: isEnabled"
);
```
What do you think?

Reply:
I'm not sure how to judge the naming of this things. To me, it was the most comfortable to name enums FunctionNameError. Are there any general recommendations about that?